### ISSUE TITLE
How to create more than 3 Input Params

### ISSUE THREAD

**Issue by Beamhi** [xingren23/ComfyFlowApp#35]
How can I create more input params when configuring a workflow UI?

**Comment by xingren23** [comment]
I want ComfyFlow App to be simple and easy for users, so I limited the options to three inputs at most. The developers optimize the default settings through testing, which is better than letting users choose, especially for regular users who don't understand sd and comfyui.

**Comment by frankyifei** [comment]
For example, I am working on a workflow which upscale the result from Ksampler. if I set positive and negative prompt, then I want to choose checkpoint, that's 3 inputs already. what about other options such as aspect ratio, or enlarge ratio. Better to let users to decide.

**Comment by xingren23** [comment]
When a creator wants to develop a workflow into a web application, they should start by putting aside the workflow itself and consider what conditions to use from the user's perspective. Imagine that the user is not aware of "Ksample," doesn't understand the difference between positive and negative prompts, and is also unfamiliar with the characteristics of various models. Exposing these complexities to the user can lead to confusion, and this is precisely the issue that Comfyflowapp aims to address.

Currently, it is not recommended to expose the aspect ratio or enlarge ratio parameters to the user.
these parameters (eg: aspect ratio or enlarge ratio) are required for every workflow, and perhaps there could be better solutions.

**Comment by SebSaga** [comment]
I'd also like to be able to adjust the number of input (and output!) parameters. I noticed it's very easy to on the ComfyFlowApp website (literally a "+" icon) and was hoping there was something similar in the local version but there isnt.

My use case is actually for allowing myself to run the web app locally so I can generate on my phone from the couch LOL 

I know what I'm doing in comfy (well, for the most part) and I'd like to be able to expand the web apps to accommodate my more advanced workflows. 

I dug through the code and figured out a way to manually update specific blocks of code in new_app.py (chatGPT helped), but there has to be a better way.

**Comment by xingren23** [comment]
yes,  creator could create ComfyFlow App with more input parameters online, https://comfyflow.app

I will add this feature to local version in the next version.

**Comment by gzjianzi** [comment]
@xingren23 
> yes, creator could create ComfyFlow App with more input parameters online, https://comfyflow.app
> 
> I will add this feature to local version in the next version.

请问开源版本还会继续更新吗？支持更多的输入参数。
谢谢
